lib/fxa-geodb.js

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var DEFAULTS = require('./defaults');
+var ERRORS = require('./errors');
+var maxmind = require('maxmind');
+var Promise = require('bluebird');
+
+module.exports = function (options) {
+  'use strict';
+
+  options = options || {};
+  var dbPath = options.dbPath || DEFAULTS.DB_PATH;
+
+  var dbLookup;
+  // we quit if the db did not load for some reason
+  try {
+    dbLookup = maxmind.open(dbPath);
+  } catch (err) {
+    // if it failed with primary database
+    // then quit with error
+    throw ERRORS.UNABLE_TO_OPEN_FILE;
```

Comment:
From mtg with Shane: fail silently until the lookup is called, and reject with a promise there if it was null.